bitcoin/src/blockdata/block.rs

```diff
@@ -276,14 +276,16 @@ impl Block {
     }
 
     /// base_size == size of header + size of encoded transaction count.
-    fn base_size(&self) -> usize { 80 + VarInt::from(self.txdata.len()).len() }
+    fn base_size(&self) -> Weight {
+        Weight::from_wu_usize(80 + VarInt::from(self.txdata.len()).len())
+    }
 
     /// Returns the size of the block.
     ///
     /// size == size of header + size of encoded transaction count + total size of transactions.
     pub fn size(&self) -> usize {
         let txs_size: usize = self.txdata.iter().map(Transaction::size).sum();
-        self.base_size() + txs_size
+        self.base_size().to_wu() as usize + txs_size
```

Comment:
Perhaps add something like `// base_size is maximum 80 + 9 so ok to cast`

Reply:
Wouldn't that comment go on L:280?

 `80 + VarInt::from(self.txdata.len()).len()`